tests/manage/z_cluster/test_scc.py

```diff
@@ -0,0 +1,234 @@
+
+        # create service account
+        service_account_obj = service_account_factory(project=project)
+
+        # create simple-app DeploymentConfig
+        simple_app_dc = pod_factory(
+            pvc=pvc,
+            pod_dict_path=constants.SIMPLE_APP_POD_YAML,
+            deployment_config=True,
+            service_account=service_account_obj,
+            security_context={"fsGroupChangePolicy": "OnRootMismatch"},
+            pod_name="simple-app",
+        )
+        logger.info(simple_app_dc.get())
+        simple_app_dc_obj = Deployment(**simple_app_dc.get())
+        simple_app_pod = Pod(
+            **get_pods_having_label(
+                label="app=simple-app", namespace=project.namespace
+            )[0]
+        )
+
+        return simple_app_dc_obj, simple_app_pod, pvc.backed_pv_obj
+
+    def test_fsgroupchangepolicy_when_depoyment_scaled(self, setup):
+        """
+        To test if any permission change/delay seen reconcile when app pod deployment with huge dumber of
+        object files and SCC setting 'fsGroupChangePolicy: OnRootMismatch'  are scaled down/up.
+        """
+        permission_map = {"2770": "", "0770": "", "0775": "", "2755": "", "0755": ""}
+        timeout = 300
+
+        # run simple-app deployment
+        simple_app_dc, simple_app_pod, pv = setup
+
+        # create objects under performance directory
+        cmd = "cd performance && for i in $(seq 0 1000000);do dd if=/dev/urandom of=object_$i bs=512 count=1;done"
+        simple_app_pod.exec_sh_cmd_on_pod(command=cmd, timeout=5400)
+
+        # get node where the simple-app pod scheduled
+        node = get_pod_node(simple_app_pod).name
+        logger.info(f"{simple_app_pod.name} pod is scheduled on node {node}")
+
+        # get the node mount
+        node_mount = (
+            OCP()
+            .exec_oc_debug_cmd(node=node, cmd_list=[f"df -h | grep {pv.name}"])
+            .split()[5]
+        )
+        logger.info(f"Node Mount: {node_mount}")
+
+        total_time = 0
+        for index, mode in enumerate(permission_map.keys()):
+            logger.info(f"PERMISSION: {mode}")
+            cmd = f"chmod {mode} {node_mount}"
+            OCP().exec_oc_debug_cmd(node=node, cmd_list=[cmd])
+
+            cmd = f"ls -ld {node_mount}"
+            mode_before = OCP().exec_oc_debug_cmd(node=node, cmd_list=[cmd]).split()[0]
+            logger.info(
+                f"Node mount permission before simple-app deployment scaling: {mode_before}"
+            )
+
+            # scale down simple-app deployment to 0
+            simple_app_dc.scale(replicas=0)
+
+            assert (
+                simple_app_dc.replicas == 0
+            ), "Failed to scale down simple-app deployment"
+
+            while OCP(kind="pod").get(
+                resource_name=simple_app_pod.name, dont_raise=True
+            ):
+                logger.info(f"{simple_app_pod.name} is still terminating; Retrying")
+
```

Comment:
Does the command keep trying until the pod disappears?

Reply:
yes

Reply:
but in the next commit i will add maxium retries so that it wont be stuck in loop forever if something goes wrong